lua/neotest-rust/init.lua

```diff
@@ -31,6 +31,11 @@ local cargo_metadata = setmetatable({}, {
     end,
 })
 
+---Find the project root directory given a current directory to work from.
+---Should no root be found, the adapter can still be used in a non-project context if a test file matches.
+---@async
```

Comment:
what makes this async? `cargo_metadata()`?

Reply:
I copied interface definitions from: https://github.com/nvim-neotest/neotest/blob/master/lua/neotest/adapters/interface.lua

Applying `@async` will give additional diagnostics: https://github.com/LuaLS/lua-language-server/wiki/Diagnostics#await

But I think in this case it is not necessary indeed